src/Planners/src/UnicycleTrajectoryGenerator.cpp

```diff
@@ -0,0 +1,844 @@
+                                                         .rightFootTrajectory.mixedVelocity,
+                                                     trajectory.rightFootMixedVelocity,
+                                                     mergePoint);
+    Planners::UnicycleUtilities::appendVectorToDeque(unicycleTrajectoryPlanner.getOutput()
+                                                         .rightFootTrajectory.mixedAcceleration,
+                                                     trajectory.rightFootMixedAcceleration,
+                                                     mergePoint);
+
+    // get steps
+    auto newLeftSteps = unicycleTrajectoryPlanner.getOutput().steps.leftSteps;
+    auto newRightSteps = unicycleTrajectoryPlanner.getOutput().steps.rightSteps;
+
+    // merge steps
+    BipedalLocomotion::Planners::UnicycleUtilities::mergeSteps(newLeftSteps,
+                                                               trajectory.leftSteps,
+                                                               time);
+    BipedalLocomotion::Planners::UnicycleUtilities::mergeSteps(newRightSteps,
+                                                               trajectory.rightSteps,
+                                                               time);
+
+    // get merge points
+    std::vector<size_t> mergePoints;
+    mergePoints = unicycleTrajectoryPlanner.getOutput().mergePoints;
+
+    // update mergePoints of new trajectory based on where the new trajectory is merged
+    for (auto& element : mergePoints)
+    {
+        element += mergePoint;
+    }
+    trajectory.mergePoints.assign(mergePoints.begin(), mergePoints.end());
+
+    // pop the first merge point
+    trajectory.mergePoints.pop_front();
+
+    return true;
+}
+
+bool BipedalLocomotion::Planners::UnicycleTrajectoryGenerator::Impl::advanceTrajectory()
+{
+
+    constexpr auto logPrefix = "[UnicycleTrajectoryGenerator::Impl::advanceTrajectory]";
+
+    // check if vector is not initialized
+    if (trajectory.leftFootinContact.empty() || trajectory.rightFootinContact.empty()
+        || trajectory.isLeftFootLastSwinging.empty() || trajectory.dcmPosition.empty()
+        || trajectory.dcmVelocity.empty() || trajectory.comPosition.empty()
+        || trajectory.comVelocity.empty() || trajectory.comAcceleration.empty())
+
+    {
+        log()->error(" {} Cannot advance empty trajectory signals.", logPrefix);
+        return false;
+    }
+
+    bool rightWasInContact = trajectory.rightFootinContact.front();
+    trajectory.rightFootinContact.pop_front();
+    trajectory.rightFootinContact.push_back(trajectory.rightFootinContact.back());
+
+    bool leftWasInContact = trajectory.leftFootinContact.front();
+    trajectory.leftFootinContact.pop_front();
+    trajectory.leftFootinContact.push_back(trajectory.leftFootinContact.back());
+
+    trajectory.isLeftFootLastSwinging.pop_front();
+    trajectory.isLeftFootLastSwinging.push_back(trajectory.isLeftFootLastSwinging.back());
```

Comment:
maybe, since I should have a template function, I can define a new static member function called something like `unrollTrajectory`, instead of a lambda function

Reply:
You can use ``auto`` in a lambda anyhow to avoid using templates. I was thinking something like
```cpp
auto trajectoryStepLambda = [](auto& inputDeque){deque.pop_front(); deque.push_back(deque.back());};
```

Reply:
Done with https://github.com/ami-iit/bipedal-locomotion-framework/pull/845/commits/6285fdabd5e555a4f84526c9ee4c49b20c4bbeb4